### ISSUE TITLE
Design - Revise search result filters

### ISSUE THREAD

**Issue by krisdelacruz** [USStateDept/State-TalentMAP#999]
## Story
As a Bidder
I want to narrow down my search results
So that I can find positions that work for my specific circumstances

### Primary 
Tarek is looking for a new position as a medical specialist, grade 2, at a domestic post.

### Secondary
Jenny's tour of duty is ending in 18 months. She needs a fair share post, ideally a place where she can use her French. However, since she has kids, she wants to make sure it's not dangerous and that she can be there for a relatively long tour.

## Findings
Usability testing indicates that the most useful filters are
- skill code
- grade
- location (post)
- bureau

However, from other background we know that secondary criteria could include: 
- tour of duty length
- danger pay
- COLA
- post differential
- projected vacancy date

## Currently 
We have most of the possible filters implemented (except for projected vacancy date). However, the UI controls need some UX and visual design iteration.  

#### All the filters
Consider 
- breaking the high priority filters out of their accordions
- turning differential into a numeric range of options rather than a binary (since it is a proxy for fairshare) 
- removing COLA (for the moment) 
- moving "domestic" under "post" 

<img width="232" alt="all the filters" src="https://user-images.githubusercontent.com/10067318/35590774-acf2c602-05bc-11e8-8ed4-d488b11ed250.png">

#### Skill code
Currently, skill codes are in an alphabetically ordered list. That flat organization does not correspond to the way that skill codes are bundled into career tracks. Consider creating a nested list under career tracks to make it easier for specialists to check off all that apply.  

<img width="239" alt="skill code filter" src="https://user-images.githubusercontent.com/10067318/35590785-bbb06cd0-05bc-11e8-88d9-bcdf39a90664.png">

#### Posts
Posts are a subset of missions. Consider combining into a nested list with the mission data. 
<img width="237" alt="post filter" src="https://user-images.githubusercontent.com/10067318/35590870-fa26d5b2-05bc-11e8-9796-d740c658d4a5.png">

#### Bureaus
<img width="238" alt="bureau filter" src="https://user-images.githubusercontent.com/10067318/35591381-a12a636e-05be-11e8-8fdf-8cd351556f8c.png">

#### Tour of duty
How can we simplify this?

<img width="239" alt="screen shot 2018-01-30 at 12 37 26 pm" src="https://user-images.githubusercontent.com/10067318/35654040-e37e2282-069f-11e8-8e81-ea2e8a1c4f3a.png">

### Lower priority
Take these filters on only after getting to a solid resolution on the core set.

#### Language
Language is a very complex filter. Let's consider whether we need it at all. 
<img width="241" alt="language filter" src="https://user-images.githubusercontent.com/10067318/35637012-c030fd82-0667-11e8-8b2c-6a8203b8d2d4.png">

#### Mission
Mission and post often overlap, particularly in small missions that only have a single post. 
<img width="237" alt="mission filter" src="https://user-images.githubusercontent.com/10067318/35637084-fd624756-0667-11e8-9fa4-198fadc6efe5.png">

### Desired
- Consistent typography across filters
- Filter control UI component match data type 
- More easily navigated long lists of options
- Combine overlapping data filters  

### Acceptance criteria
Approval at one of the regular approval sessions
Usability testing shows FS employees can use the filters
Documented styles and UI specs ready for implementation

**Comment by esgoodman** [comment]
Note: `skill code` needs more research to clarify whether it refers to the title of the skill code or the more general cone to which the skill code belongs.

**Comment by wongperson** [comment]
Design-centric task, no development requested just yet.

**Comment by wongperson** [comment]
@esgoodman to revise this story prior to planning 1/31/18

**Comment by krisdelacruz** [comment]
@esgoodman This was a research task. We have a card sorting research plan that we were going to execute utilizing Optimal Sort but it got de-prioritized. We wanted to test two things:

1. Which location filter is being utilized more (Post, Mission, Bureau)?
2. If there is a logical method user are drilling down to them?

**Comment by esgoodman** [comment]
Here is how USAJobs does some of this. Something to take into account!

#### Screen shot skill code filter
![screen shot skill code filter](https://user-images.githubusercontent.com/10067318/35761989-53d3e684-0844-11e8-88e3-9737065b4f79.png)

#### Screen shot grade filter

![screen shot grade filter](https://user-images.githubusercontent.com/10067318/35761990-55ee6296-0844-11e8-9b9b-4691e0231fc3.png)

#### Screen shot department and agency filter

![screen shot department filter ](https://user-images.githubusercontent.com/10067318/35761991-5810c686-0844-11e8-9910-790dc8468e5a.png)

**Comment by wongperson** [comment]
Split into separate issues, one per filter. Task for Edwin to take so that each task is ennumerated for front end dev.
